.github/workflows/scripts/summarize_testng.py

```
@@ -37,66 +47,184 @@ def _arg(flag, default):
+        "passed": c.get("PASS", 0),
+        "failed": failed,
+        "skipped": c.get("SKIP", 0),
+        "raw_total": raw_total,
+        "retries": max(0, raw_total - total),
+        "known": known,
+        "regressions": failed - known,
+        "fails_by_class": fails_by_class,
+    }
+
+
+def _md_cell(text):
```

Ruff (0.16.0)

[warning] Missing return type annotation for private function `_md_cell`

(ANN202)